File: adventure-client/src/main/java/com/adam/adventure/render/ui/NiftyUiRenderable.java
package com.adam.adventure.render.ui;

import com.adam.adventure.render.Renderer;
import com.adam.adventure.render.renderable.Renderable;
import de.lessvoid.nifty.Nifty;

public class NiftyUiRenderable implements Renderable {

    private final Nifty nifty;

    NiftyUiRenderable(final Nifty nifty) {
        this.nifty = nifty;
    }

    @Override
    public int getZIndex() {
        //Always at the top
        return Integer.MAX_VALUE;
    }

    @Override
    public void initialise(final Renderer renderer) {
        //Nothing needs doing
    }

    @Override
    public void prepare(final Renderer renderer) {
        nifty.update();
    }

    @Override
    public void render(final Renderer renderer) {
        nifty.render(false);
    }

    @Override
    public void after(final Renderer renderer) {
        // Nothing needs doing
    }
}

File: adventure-client/src/main/java/com/adam/adventure/render/renderable/NiftyUiRenderable.kt
package com.adam.adventure.render.renderable

import com.adam.adventure.render.Renderer
import de.lessvoid.nifty.Nifty

class NiftyUiRenderable(val nifty : Nifty) : Renderable {
    override fun getZIndex(): Int {
        return Integer.MAX_VALUE
    }


    override fun render(renderer: Renderer?) {
        nifty.render(false)
    }
}